In [1]:
import numpy as np

__Задача 1__

##### Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Составим функцию для нахождения всех возможных комбинации данных

$
C_{n}^k = \cfrac{n!}{k!(n - k)!}
$

In [2]:
n = 52 # общее число карт
k = 4 # вытянутые 4 карты
abcd = np.arange(1,5)
abcd

array([1, 2, 3, 4])

In [3]:
def combinations(n: int, k: int):
    '''число сочетаний'''
    
    factorial = np.math.factorial
    result = factorial(n) // (factorial(k) * factorial(n - k))
    
    return result

In [4]:
all_varies = combinations(n, k) # все вариенты вытащить 4 карты из 52 (для себя)
all_varies

270725

1) число способов достать все возможные комбинации тузов из 4 карт

In [5]:
all_of_four = [combinations(4, i) for i in abcd] # все вариенты вытащить тузы карты из 4 тузов
all_of_four

[4, 6, 4, 1]

Мы видим, что есть только 1 исход при котором все 4 туза вытащины из колоды

2) число способов достать все возможные комбинации оставшиеся не тузы из оставшихся в колоде карт

In [6]:
remaining = [combinations(n-4, 4-i) for i in abcd] # все варианты вытащить оставныеся не тузы из оставшейся колоды
remaining

[17296, 1128, 48, 1]

Так же видим что есть только 1 исход, при котором не тузы не вытащены из колоды

3) Найдём произведения каждого варианта вытащенных тузов и остальных карт

In [7]:
abcd_varies = [all_of_four[i] * remaining[i] for i in range(k)] 
# все варианты вытащенных 4 карты для разного колическа вытащенных тузов из 52
abcd_varies

[69184, 6768, 192, 1]

4) Суммируем все варианты - получаем ответ

In [8]:
sum(abcd_varies)

76145

Ответ:

Число исходов, при которых вытащен козябы 1 туз из 4 вытянутых карт из колоды в 52 карты, равно 76145.

__Задача 2__

##### Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

Есть только 1 массив - 7 человек;

Есть только 1 исход - определённые люди должны сесть рядом:

    а) 2 человека

Допусти что 6 человек уже заняли свои места на скамейке, среди них есть 1й определённый человек, а 2й определённый человек ещё не сел на скомейку. Учитывая, что мы не знаем где сидит 1й - так как люди содятся случайным образом, мы можем предположить, что 2й человек может сесть как справа так и слева от каждого уже сидящего человека, то есть $\frac{2}{1}$, но так как сидящич людей уже 6 а не 1, то получается $\frac{2}{(1+6)} = \frac{2}{7}$.

In [9]:
f'Вероятность того, что рядом будут сидеть два определённых человека = {2/7}'

'Вероятность того, что рядом будут сидеть два определённых человека = 0.2857142857142857'

    б) три определённых человека

Берём за основу предыдущее решение. Допустим 2е людей уже сидят рядом, тогда уменьшаем количество вариентов для 3го на 1, то есть $\frac{2}{7-1}$. Так как для того, что бы все 3 определённых человека сидели рядом, нужно что бы 2 из них уже сидели рядом, а значит составляем следующую формулу $\frac{2}{7} \cdot \frac{2}{7-1} = \frac{2}{7} \cdot \frac{1}{3} = \frac{2}{7 \cdot 3} = \frac{2}{21}$.

In [10]:
f'Вероятность того, что рядом будут сидеть три определённых человека = {2/21}'

'Вероятность того, что рядом будут сидеть три определённых человека = 0.09523809523809523'

__Задача 3__

##### Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

    а) Все

1) Найдём вероятность вотянуть первый билет из тек которые студент знает - $\frac{50}{60}$.

In [11]:
al = 60 
kn = 50
un = al-k

In [12]:
t = kn/al
f'Шанс вытянуть 1й билет из тех которые студент знает = {t}'

'Шанс вытянуть 1й билет из тех которые студент знает = 0.8333333333333334'

2) Теперь найдём вероятность вытянуть 2 оставшихся билета, при том, что 1й был вытянут из тех, которые студент знает, коэффициент второго билета будет равен $\frac{50-1}{60-1}$, а 3го, при условии, что 2й тоже был из тех, которые студент знает, соответственно $\frac{50-2}{60-2}$

In [13]:
t2 = (kn-1)/(al-1)
f'Шанс вытянуть 2й билет из тех которые студент знает = {t2}'

'Шанс вытянуть 2й билет из тех которые студент знает = 0.8305084745762712'

In [14]:
t3 = (kn-2)/(al-2)
f'Шанс вытянуть 3й билет из тех которые студент знает = {t3}'

'Шанс вытянуть 3й билет из тех которые студент знает = 0.8275862068965517'

3) Зная вероятности всех трёх билетов, нужно узнать вероятность того, что все эти 3 билета будут вытянуты

In [15]:
f'Вероятность того, что все 3 вытянутых билета известны студенту = {t * t2 * t3}'

'Вероятность того, что все 3 вытянутых билета известны студенту = 0.5727644652250146'

    б) Два

1) Здесь для нас важно каким будет вытянут билет, который студент не знает, так как он всего лишь 1, а общее число вытянутых билетов 3, можно сразу сказать, что есть 3 варинта вытянуть не известный билет - 1м, 2м или 3м.

2) Если есть 1 билет из 3, который не известен студенту, то каждый из трёх билетов может оказаться либо известным, либо неизвестным. Найдём вероятности для каждого из 3х исходов:

In [16]:
r1 = (kn/al) * ((kn-1)/(al-1)) * (un/(al-2))
f'Вероятность 1го исхода (известный, известный, не известный) = {r1}'

'Вероятность 1го исхода (известный, известный, не известный) = 0.6682252094291837'

In [17]:
r2 = (kn/al) * (un/(al-1)) * ((kn-1)/(al-2))
f'Вероятность 2го исхода (известный, не известный, известный) = {r2}'

'Вероятность 2го исхода (известный, не известный, известный) = 0.6682252094291837'

In [18]:
r3 = (un/(al)) * (kn/(al-1)) * ((kn-1)/(al-2))
f'Вероятность 2го исхода (не известный, известный, известный) = {r3}'

'Вероятность 2го исхода (не известный, известный, известный) = 0.6682252094291836'

Как мы видим результаты для всех 3х исходов почти не изменились и тем не менее для числоты выведем средний результат

In [19]:
f'Вероятность того, что 2 из 3х вытянутых билетов будут известны студенту = {(r1 + r2 + r3)/3}'

'Вероятность того, что 2 из 3х вытянутых билетов будут известны студенту = 0.6682252094291837'

__Задача 4__

##### Бросается игральная кость. Пусть событие `A` - появление чётного числа, событие `B` - появление числа больше трёх. Являются ли эти события независимыми?

$
P(A) = \frac{1}{6} \\
P(B) = \frac{1}{2} \\
P(AB) = \frac{1}{12} \\
P(A|B) = \cfrac{P(AB)}{P(B)} = \cfrac{\frac{1}{12}}{\frac{1}{2}} = \frac{1 \cdot 2}{12 \cdot 1} = \frac{1}{6}
$

Поскольку  $P(A|B)=P(A) \Rightarrow A \land B$  независимы.

__Задача 5__

##### Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

##### _Вопрос_: какова вероятность, что вы действительно больны ей?

##### _Подсказка_: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

$
P(A|B) = \dfrac{P(B|A) \cdot P(A)}{P(B)} \\
P(A) = 0,1\% - вероятность\ заразиться\\
P(B|A) = 99\% - вероятность\ точности\ теста\ при\ условии\ заражения\\
P(B) = P(B|A) \cdot P(A) + P(B| \overline{A}) \cdot P (\overline{A}) - полная\ вероятность\ наличия\ болезни
$

In [92]:
A = 0.1/100
BfA = 99/100
A,BfA

(0.001, 0.99)

In [93]:
AfB = (BfA * A)/(BfA * A + (1-BfA) * (1-A))
AfB

0.09016393442622944

In [97]:
AfB2 = (BfA * AfB)/(BfA * AfB + (1-BfA) * (1-AfB))
f'вероятность, что Я действительно болен = {AfB2 * 100}%'

'вероятность, что Я действительно болен = 90.74999999999999%'